cmd/notification.go

```diff
@@ -380,7 +380,12 @@ func (sys *NotificationSys) AddRulesMap(bucketName string, rulesMap event.RulesM
 		rulesMap.Add(targetRulesMap)
 	}
 
-	rulesMap.Add(sys.bucketRulesMap[bucketName])
+	// Do not add for an empty rulesMap.
+	if len(rulesMap) == 0 {
+		delete(sys.bucketRulesMap, bucketName)
+	} else {
+		rulesMap.Add(sys.bucketRulesMap[bucketName])
+	}
 	sys.bucketRulesMap[bucketName] = rulesMap
```

Comment:
You are adding empty `rulesMap` here.  You would need to fix it properly.

Reply:
*oops . got it

Reply:
changed it @balamurugana